chains/ethereum/server/src/client.rs

```diff
@@ -151,28 +158,82 @@ where
             .as_u128())
     }
 
-    #[allow(clippy::unused_async)]
+    #[allow(clippy::unused_async, clippy::missing_errors_doc)]
     pub async fn coins(&self, _address: &Address, _block: &BlockIdentifier) -> Result<Vec<Coin>> {
         anyhow::bail!("not a utxo chain");
     }
 
-    pub async fn faucet(&self, address: &Address, param: u128) -> Result<Vec<u8>> {
-        // first account will be the coinbase account on a dev net
-        let coinbase = self.client.get_accounts().await?[0];
-        let address: H160 = address.address().parse()?;
-        let tx = TransactionRequest::new().to(address).value(param).from(coinbase);
-        Ok(self
-            .client
-            .send_transaction(tx, None)
-            .await?
-            .confirmations(2)
-            .await?
-            .context("failed to retrieve tx receipt")?
-            .transaction_hash
-            .0
-            .to_vec())
+    #[allow(clippy::single_match_else, clippy::missing_errors_doc)]
+    pub async fn faucet(
+        &self,
+        address: &Address,
+        param: u128,
+        private_key: Option<&str>,
+    ) -> Result<Vec<u8>> {
+        match private_key {
+            Some(private_key) => {
+                let rpc_url_str = "http://localhost:8547";
+                let rpc_url = Url::parse(rpc_url_str)?; //.expect("Invalid URL");
+
+                let http = Http::new(rpc_url);
+                let provider = Provider::<Http>::new(http);
+                let chain_id = provider.get_chainid().await?;
```

Comment:
This should not be hard codec here, why you can't reuse `self.client`? is it another client? if so, we need a cleaner way to support arbitrum.

Reply:
resolved